File: Code/1534/1534.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
文件名: PersonalLeetCode
作者: Bolun Xu
创建日期: 2025
版本: 1.0
描述: 统计好三元组。
时间复杂度： O(N ^ 3)
空间复杂度： O(1)

"""
from typing import List


class Solution:
    def countGoodTriplets(self, arr: List[int], a: int, b: int, c: int) -> int:
        ans = 0
        for i in range(len(arr) - 2):
            for j in range(i + 1, len(arr) - 1):
                if abs(arr[i] - arr[j]) <= a:
                    for k in range(j + 1, len(arr)):
                        if abs(arr[j] - arr[k]) <= b and abs(arr[i] - arr[k]) <= c:
                            ans += 1
        return ans


if __name__ == '__main__':
    sol = Solution()
    print(sol.countGoodTriplets(arr=[3, 0, 1, 1, 9, 7], a=7, b=2, c=3))

File: Code/1/5/1534/1534.cpp
/******************************************************************************
 * 文件名: 1534
 * 作者: BOLUN XU
 * 创建日期: 2025
 * 版本: 1.0
 * 描述: 。
 * 时间复杂度：
 * 空间复杂度：
 ******************************************************************************/
#include <iostream>
#include <vector>
#include <cmath> // for abs function

using namespace std;

class Solution {
public:
  int countGoodTriplets(vector<int>& arr, int a, int b, int c) {
    int ans = 0;
    int n = arr.size();
    for (int i = 0; i < n - 2; ++i) {
      for (int j = i + 1; j < n - 1; ++j) {
        if (abs(arr[i] - arr[j]) <= a) {
          for (int k = j + 1; k < n; ++k) {
            if (abs(arr[j] - arr[k]) <= b && abs(arr[i] - arr[k]) <= c) {
              ans++;
            }
          }
        }
      }
    }
    return ans;
  }
};

int main() {
  Solution sol;
  vector<int> arr = {3, 0, 1, 1, 9, 7};
  int a = 7, b = 2, c = 3;
  cout << sol.countGoodTriplets(arr, a, b, c) << endl; // Output: 4
  return 0;
}